internal/handlers/sqlite/msg_count.go

```diff
@@ -16,12 +16,68 @@ package sqlite
 
 import (
 	"context"
+	"errors"
 
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/iterator"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 // MsgCount implements HandlerInterface.
 func (h *Handler) MsgCount(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
-	return nil, notImplemented(must.NotFail(msg.Document()).Command())
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	params, err := common.GetCountParams(document, h.L)
+	if err != nil {
+		return nil, err
+	}
+
+	db := h.b.Database(params.DB)
+	defer db.Close()
+
+	queryRes, err := db.Collection(params.Collection).Query(ctx, nil)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	iter := queryRes.Iter
+
+	closer := iterator.NewMultiCloser(iter)
+	defer closer.Close()
+
+	iter = common.FilterIterator(iter, closer, params.Filter)
+
+	iter = common.SkipIterator(iter, closer, params.Skip)
+
+	iter = common.LimitIterator(iter, closer, params.Limit)
+
+	iter = common.CountIterator(iter, closer, "count")
+
+	_, res, err := iter.Next()
+	if errors.Is(err, iterator.ErrIteratorDone) {
+		err = nil
+	}
+
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	count, _ := res.Get("count")
```

Comment:
is the error was `ErrIteratorDone` could it be possible that `res` is `nil` here?

Reply:
Yes

Reply:
And `count` will be zero.